src/Akka.Streams.Kafka/Stages/ProducerStage.cs

```diff
@@ -97,7 +105,21 @@ private void OnProducerError(object sender, Error error)
 
             if (!KafkaExtensions.IsBrokerErrorRetriable(error) && !KafkaExtensions.IsLocalErrorRetriable(error))
             {
-                FailStage(new Exception(error.Reason));
+                var exception = new KafkaException(error);
+                switch (_decider(exception))
+                {
+                    case Directive.Stop:
+                        // Throw
+                        _completionState.TrySetException(exception);
+                        FailStage(exception);
+                        break;
+                    case Directive.Resume:
+                        // TODO: resume the stream
```

Comment:
I don't know what to do here. Kafka driver itself has a retry policy for message delivery. If I get unretrievable error, that's mean I should not try again.

Reply:
What about resetting connection (disposing old one, setting a new)?